File: ReAct_v_ts/src/status.ts
import type OpenAI from "openai";

export type AgentPhase = "THINK" | "ACT" | "OBSERVE" | "END";

export type ChatMessage =
  OpenAI.Chat.Completions.ChatCompletionMessageParam;

export type AssistantMessage =
  OpenAI.Chat.Completions.ChatCompletionMessage;

export interface PendingToolResult {
  tool_call_id: string;
  result: string;
}

export interface CheckpointData {
  task_id: string;
  step: number;
  status: AgentPhase;
  max_steps: number;
  messages: ChatMessage[];
  pending?: PendingToolResult[];
  pending_results?: PendingToolResult[];
  last_message: AssistantMessage | null;
  _updated_at?: number;
}

export class AgentStatus {
  constructor(
    public messages: ChatMessage[],
    public status: AgentPhase = "THINK",
    public step = 0,
    public maxSteps = 50,
    public lastMessage: AssistantMessage | null = null,
    public pendingResults: PendingToolResult[] = [],
  ) {}

  static fromCheckpoint(data: CheckpointData): AgentStatus {
    return new AgentStatus(
      data.messages ?? [],
      data.status ?? "THINK",
      data.step ?? 0,
      data.max_steps ?? 50,
      data.last_message ?? null,
      data.pending_results ?? data.pending ?? [],
    );
  }
}

File: ReAct_v/status.py
from openai.types.chat import ChatCompletionMessage

class AgentStatus:

    def __init__(self,
        messages,
        status="THINK",
        step=0,
        max_steps=50,
        last_message=None,
        pending_results=None,
        ):
        self.messages = messages
        self.status = status
        self.step = step
        self.max_steps = max_steps
        self.last_message = last_message
        self.pending_results = pending_results or []

    @classmethod
    def from_dict(cls, data):
        raw_last_message = data.get("last_message")
        last_message = (
            ChatCompletionMessage.model_validate(raw_last_message)
            if raw_last_message is not None
            else None
        )

        return cls(
            messages=data.get("messages", []),
            status=data.get("status", "THINK"),
            step=data.get("step", 0),
            max_steps=data.get("max_steps", 50),
            last_message=last_message,
            pending_results=data.get("pending_results", []),
        )